### PR TITLE
awsconfig: use ResourceSet & ResourceRouter

### PR DESCRIPTION
**Pull request by kopiczko**

### CODE REVIEW

File: service/awsconfig/framework.go [hunk 1]
```diff
@@ -451,19 +445,92 @@ func newVersionedResources(config FrameworkConfig) (map[string][]framework.Resou
...
+	}
+
+	var resourceSetV1 *framework.ResourceSet
+	{
+		c := framework.ResourceSetConfig{
+			Handles: handlesFuncFunc([]string{
+				"",
+				"0.1.0",
+				"1.0.0",
+			}),
```
Comment (kopiczko) [inline service/awsconfig/framework.go]: Please check those values carefully.

File: service/awsconfig/framework.go [hunk 2]
```diff
@@ -451,19 +445,92 @@ func newVersionedResources(config FrameworkConfig) (map[string][]framework.Resou
...
+		}
+	}
+
+	var resourceSetV2 *framework.ResourceSet
+	{
+		c := framework.ResourceSetConfig{
+			Handles: handlesFuncFunc([]string{
+				"0.2.0",
+				"2.0.0",
+			}),
```
Comment (kopiczko) [inline service/awsconfig/framework.go]: Please check those values carefully.

File: service/awsconfig/framework.go [hunk 3]
```diff
@@ -451,19 +445,92 @@ func newVersionedResources(config FrameworkConfig) (map[string][]framework.Resou
...
+	}
+
+	var resourceSetV4 *framework.ResourceSet
+	{
+		c := framework.ResourceSetConfig{
+			Handles: handlesFuncFunc([]string{
+				"2.0.1",
+				// 2.0.2 fixes missing region in host account credentials, the change only affects service/framework.go
+				"2.0.2",
+			}),
```
Comment (kopiczko) [inline service/awsconfig/framework.go]: Please check those values carefully.

File: service/awsconfig/framework.go [hunk 4]
```diff
@@ -451,19 +445,92 @@ func newVersionedResources(config FrameworkConfig) (map[string][]framework.Resou
...
+
+			for _, v := range handledVersionBundles {
+				if versionBundleVersion == v {
+					return true
+				}
+			}
+
+			return false
+		}
+	}
```
Comment (kopiczko) [inline service/awsconfig/framework.go]: This is a little weird. But it'll got to separate v* packages like in kvm-operator. #TransitionPeriod

File: service/awsconfig/framework.go [hunk 5]
```diff
@@ -451,19 +445,92 @@ func newVersionedResources(config FrameworkConfig) (map[string][]framework.Resou
...
+
+	var resourceSetV1 *framework.ResourceSet
```
Comment (kopiczko) [inline service/awsconfig/framework.go]: I named that v1 - this is the name of package where leagacy resource will spend the rest of its life.

File: service/awsconfig/framework.go [hunk 6]
```diff
@@ -451,19 +445,92 @@ func newVersionedResources(config FrameworkConfig) (map[string][]framework.Resou
...
+	}
+
+	var resourceRouter *framework.ResourceRouter
+	{
+		c := framework.ResourceRouterConfig{}
+
+		c.ResourceSets = []*framework.ResourceSet{
+			resourceSetV1,
+			resourceSetV2,
+			resourceSetV4,
```
Comment (kopiczko) [inline service/awsconfig/framework.go]: As you can see there is no v3, I'll remove the directory later after copying stuff to v4.

File: service/awsconfig/framework.go [hunk 7]
```diff
@@ -451,19 +445,92 @@ func newVersionedResources(config FrameworkConfig) (map[string][]framework.Resou
...
+		}
+	}
+
+	var resourceSetV4 *framework.ResourceSet
+	{
+		c := framework.ResourceSetConfig{
+			Handles: handlesFuncFunc([]string{
+				"2.0.1",
+				// 2.0.2 fixes missing region in host account credentials, the change only affects service/framework.go
+				"2.0.2",
```
Comment (kopiczko) [inline service/awsconfig/framework.go]: BTW I don't understand the purpose of 2.0.2. It was bug fix in the operator. I'd stay we shouldn't introduce a new version for that.

File: service/awsconfig/framework.go (hunk 6 above)
Comment (xh3b4sd) [reply service/awsconfig/framework.go]: Why is this though? This seems to be super wrong.

File: service/awsconfig/framework.go (hunk 4 above)
Comment (rossf7) [reply service/awsconfig/framework.go]: Why is the var named `kvmConfig`? I'm good with #TransitionPeriod but that's weird!

File: service/awsconfig/framework.go (hunk 5 above)
Comment (rossf7) [reply service/awsconfig/framework.go]: Very nice, I like that we keep legacy isolated in its final weeks!

File: service/awsconfig/framework.go (hunk 1 above)
Comment (rossf7) [reply service/awsconfig/framework.go]: These are correct. `0.1.0` was the initial legacy release. We changed to `1.0.0` for semver.

File: service/awsconfig/framework.go (hunk 7 above)
Comment (rossf7) [reply service/awsconfig/framework.go]: You're right `2.0.2` was a bug fix to the operator to add the missing host region flag. Adding a new version confused things.

File: service/awsconfig/framework.go (hunk 6 above)
Comment (rossf7) [reply service/awsconfig/framework.go]: I think we got out of sync due to the CRD  / CF migration. e.g we had 

- cloudconfigv1 - TPR
- cloudconfigv2 - CRD
- cloudconfigv3 - CRD + k8scloudconfig v_3_0_0 for CloudFormation

Could we have resourceSetV3 supporting bundles `2.0.1` and `2.0.2`? That makes more sense to me.

Comment (kopiczko) [reply service/awsconfig/framework.go]: I think naming tired to reflect version bundle versions and that's why we have situation like that. But we don't have to worry, as the new structure is there to improve.

File: service/awsconfig/framework.go (hunk 4 above)
Comment (kopiczko) [reply service/awsconfig/framework.go]: I should be awsConfig, thanks!

File: service/awsconfig/framework.go (hunk 6 above)
Comment (kopiczko) [reply service/awsconfig/framework.go]: I would flatten v3 and v4. WDYT?

Comment (rossf7) [reply service/awsconfig/framework.go]: I think flattening is a good idea. The problem will go away as we delete old resource sets but in the short term it's confusing.

Comment (kopiczko) [reply service/awsconfig/framework.go]: Renamed to resourceSetV3
I'll flatten v4 to v3 in follow up PRs.

File: service/awsconfig/framework.go (hunk 4 above)
Comment (kopiczko) [reply service/awsconfig/framework.go]: Done.

**Comment by kopiczko** [comment]
PTALA

File: service/awsconfig/framework.go [hunk 8]
```diff
@@ -451,19 +445,92 @@ func newVersionedResources(config FrameworkConfig) (map[string][]framework.Resou
...
+		}
+	}
+
+	var resourceRouter *framework.ResourceRouter
+	{
+		c := framework.ResourceRouterConfig{
+			ResourceSets: []*framework.ResourceSet{
+				resourceSetV1,
+				resourceSetV2,
+				resourceSetV3,
```
Comment (rossf7) [inline service/awsconfig/framework.go]: <3

**Pull request merged**